share/smack/top.py

```diff
@@ -305,6 +314,12 @@ def build_libs(args):
   if args.pthread:
     libs += ['pthread.c']
 
+  if args.memory_safety:
```

Comment:
Is there a particular reason that the definitions are linked only when memory safety is being checked? We need them to check signed integer overflows, too. Can we always link them?

Reply:
My worry is that linking these stubs may be a negative impact on verification scalability. I am supposing that many programs/assertions do not rely, e.g., on the details of string operations. Therefore I am in favor of conditional linkage. Admittedly, I have no data to support this opinion :-)

One solution to your problem is to link this for signed integer overflows too. Please feel free add that as an enabling condition.

In the long run though, I think a more generic mechanism for linking stubs will be useful/necessary. For instance, besides enabling or disabling a single predefined stub for `string.h` functions, we might want to choose between different stubs of the same `string.h` functions, e.g., which have weaker specifications yet are easier for the verifier to reason about. Or, for instance, we may want to link user-defined stubs.

Reply:
For now, I agree with @michael-emmi : let's just add integer overflow flag in addition to memory safety to this check. @shaobo-he : can you please do this ASAP and then test this branch again?

In the long run, we could introduce a command like option called `link-models` or something like that, which we would use to select which models are linked in. Something like `link-models=pthreads,string` would link in the respective models. Then for SVCOMP we can enable those if particular flags are selected. I will create a separate issue for this.

Reply:
Done.